commands/xinfo-groups.md

```diff
@@ -6,29 +6,68 @@ By default, only the following information is provided for each of the groups:
 * **consumers**: the number of consumers in the group
 * **pending**: the length of the group's pending entries list (PEL), which are messages that were delivered but are yet to be acknowledged
 * **last-delivered-id**: the ID of the last entry delivered the group's consumers
+* **last-delivered-offset**: the offset of the last entry delivered to group's consumers
+* **lag**: the number of entries in the stream that are still waiting to be delivered to the group's consumers, or a NULL when that number can't be determined.
+
+### Consumer group lag
+
+The lag of a given consumer group is the number of entries in the range between the group's `last-delivered-id` and the stream's `last-generated-id`.
+Put differently, it is the number of entries that are yet to be delivered to the group's consumers.
+
+The values and trends of this metric are helpful in making scaling decisions about the consumer group.
+You can address high lag values by adding more consumers to the group, whereas low values may indicate that you can remove consumers from the group to scale it down.
+
+Redis reports the lag of a consumer group by keeping two offsets: one for the stream and another one for the consumer group.
+The lag is the difference between these two offsets.
+
+The stream's offset (the `last-offset` field of the `XINFO STREAM` command) is incremented by one with every `XADD` and counts all of the entries added to the stream during its lifetime.
+
+The consumer group's offset, `last-delivered-offset`, is the offset of the entry referenced by the group's `last-delivered-id`.
+
+There are two special cases in which this mechanism is unable to report the lag:
+
+1. A consumer group is created or set with an arbitrary last delivered ID (the `XGROUP CREATE` and `XGROUP SETID` commands, respectively). An arbitrary ID is any ID that isn't the ID of the stream's first entry, its last entry or the zero ("0-0") ID.
+2. One or more entries between the group's `last-delivered-id` and the stream's `last-generated-id` were deleted (with `XDEL` or a trimming operation).
+
+In both cases, the group's offset is considered invalid, and the returned value is set to NULL to signal that the lag isn't currently available.
+
+However, the lag's unavailability is only temporary and resolves itself during regular operation.
```

Comment:
Unavailability is an unusual word. It's extra unusual that it resolves itself. :-) Maybe this sentence can be made a little lighter? Ideas:

*  "the lag's unavailability" => "the unavailability of the lag" or "absence of the lag" or "the lag is only temporarily missing"
* "resolves itself" => "becomes available again automatically"
* Split into two sentences

Reply:
Resolved.